### PR TITLE
Auth and product creation

### PR DESCRIPTION
**Pull request by VanishMax**
Implemented features of connecting client and server sides of Innopoints:
* Authentication
* Drag&Drop in the Dropzone
* Image (Not always working) uploading in product creation
* Saving draft of the product
* Validation and product creation
* Displaying store items (simple preload() example)

### CODE REVIEW

File: src/store/item.js [hunk 1]
```diff
@@ -31,52 +35,124 @@ const newItem = () => ({
...
+            images: imgs,
+            amount: variety.sizes[size],
+            size,
+          });
+        }
+      }
+    } else {
+      varieties.push({
+        color: variety.color.replace(/[^0-9a-f]/gi, ''),
+        images: variety.images.map(img => img.url),
```
Comment (aabounegm) [inline src/store/item.js]:
```suggestion
        images: imgs,
```

File: src/containers/profile/info.svelte [hunk 2]
```diff
@@ -2,24 +2,21 @@
...
-  }
+  };
 </script>
 
 <section class="quick-info padded">
   <div class="personal">
-    <h1 class="btn-align">{$user.name} {$user.surname}</h1>
+    <h1 class="btn-align">{$user.full_name}</h1>
     {#if tgChange}
       <div class="telegram input">
```
[the remark this answers is not in the document]
Comment (illright) [reply src/containers/profile/info.svelte]: Ugh. God damn this error hint. Guess I forgot to fix it. Stay put

[the remark this answers is not in the document]
Comment (illright) [reply src/containers/profile/info.svelte]: Done

File: src/containers/profile/info.svelte [hunk 3]
```diff
@@ -2,24 +2,21 @@
...
 
-  function toggleTgChange(bool) {
-    tgChange = bool;
-  }
-
-  function save() {
-    if (tgVal !== '') changeUserField('telegram', tgVal);
+  const toggleTgChange = (bool) => tgChange = bool;
+  const save = () => {
+    if (tgVal !== '') changeTelegram(tgVal);
```
[the remark this answers is not in the document]
Comment (illright) [reply src/containers/profile/info.svelte]: Handle it here

File: src/components/header.svelte [hunk 4]
```diff
@@ -1,20 +1,27 @@
...
-        <svg src="images/icons/user.svg" />
-      </Button>
+      {#if isProfile}
+        <NotificationCenter />
+        <Button on:click={signOut}>sign out</Button>
+      {:else}
+        <Button href="/profile" isRound>
+          <svg src="images/icons/user.svg" />
+        </Button>
+      {/if}
```
[the remark this answers is not in the document]
Comment (illright) [reply src/components/header.svelte]: As for signing out, how often do you do that on a university portal?
About the notifications, @VanishMax, maybe the markup was a bit misleading, but the bell must be present in the header for all pages

[the remark this answers is not in the document]
Comment (VanishMax) [reply src/components/header.svelte]: Got it

File: src/utils/request.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import {API_HOST} from '@/constants/env';
+
+// fetchType is needed for server-side request:
+// in Sapper's preload() function there's this.fetch function that
+// distinguishes between node-fetch and usual fetch depending on env
+export default async (fetchType = fetch, url = '', method = 'GET', body = {}, isForm = false) => {
+  return await fetchType(API_HOST + url, {
```
[the remark this answers is not in the document]
Comment (VanishMax) [reply src/utils/request.js]: Yep, my fault with the `logout` example. Fixed

File: src/utils/request.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    credentials: 'include',
+    body: isForm ? body : Object.entries(body).length ? JSON.stringify(body) : undefined,
+    headers: {
+      'content-type': (!isForm && Object.entries(body).length) ? 'application/json' : undefined,
+    },
+    method,
+  }).then(res => {
```
[the remark this answers is not in the document]
Comment (VanishMax) [reply src/utils/request.js]: just let it be for now

File: src/store/item.js (hunk 1 above)
Comment (VanishMax) [reply src/store/item.js]: Nope, backend needs only uri strings, not objects

File: src/containers/profile/info.svelte (hunk 3 above)
[the remark this answers is not in the document]
Comment (VanishMax) [reply src/containers/profile/info.svelte]: In the future: when we'll connect profile's frontend with the server

**Pull request merged**